src/overlays/actors/ovl_En_Bombf/z_en_bombf.c

```diff
@@ -34,53 +35,470 @@ const ActorInit En_Bombf_InitVars = {
+            }
+            Actor_PlaySfxAtPos(&this->actor, NA_SE_EV_BOMB_BOUND);
+            Actor_MoveWithGravity(&this->actor);
+            DREG(6) = 1;
+            Actor_UpdateBgCheckInfo(globalCtx, &this->actor, 5.0f, 10.0f, 0.0f, 0x1F);
+            DREG(6) = 0;
+            this->actor.speedXZ *= 0.7f;
+            this->actor.bgCheckFlags &= ~8;
+        }
+
+        if ((this->colliderCylinder.base.acFlags & AC_HIT) ||
+            ((this->colliderCylinder.base.ocFlags1 & OC1_HIT) && (this->colliderCylinder.base.oc->category == 5))) {
+            this->unk_1F8 = 1;
+            this->timer = 0;
+        } else if ((this->timer > 100) && func_80123F48(globalCtx, &this->actor.world.pos, 30.0f, 50.0f)) {
+            this->timer = 100;
+        }
+
+        if (this->unk_1F8 != 0) {
+            sp5C.y = 0.2f;
+            sp68 = this->actor.world.pos;
+            sp68.y += 25.0f;
+
+            if (this->timer < 0x7F) {
+                if ((globalCtx->gameplayFrames % 2) == 0) {
+                    EffectSsGSpk_SpawnFuse(globalCtx, &this->actor, &sp68, &sp8C, &sp74);
+                }
+                Actor_PlaySfxAtPos(&this->actor, NA_SE_IT_BOMB_IGNIT - SFX_FLAG);
+                sp68.y += 3.0f;
+                func_800B0DE0(globalCtx, &sp68, &sp8C, &sp5C, &sp58, &sp58, 0x32, 5);
+            }
+
+            if ((this->timer == 3) || (this->timer == 30) || (this->timer == 50) || (this->timer == 70)) {
+                this->unk_1FE = this->unk_1FE >> 1;
+            }
+
+            if ((this->timer < 100) && (this->timer & (this->unk_1FE + 1))) {
+                Math_SmoothStepToF(&this->unk_200, 150.0f, 1.0f, 150.0f / this->unk_1FE, 0.0f);
+            } else {
+                Math_SmoothStepToF(&this->unk_200, 0.0f, 1.0f, 150.0f / this->unk_1FE, 0.0f);
+            }
+
+            if (this->timer < 3) {
+                Actor_SetScale(&this->actor, this->actor.scale.x + 0.002f);
+            }
+
+            if (this->timer == 0) {
+                sp68 = this->actor.world.pos;
+                sp68.y += 10.0f;
+
+                if (Actor_HasParent(&this->actor, globalCtx)) {
+                    sp68.y += 30.0f;
+                }
+
+                Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_CLEAR_TAG, sp68.x, sp68.y, sp68.z, 0, 0, 0,
+                            CLEAR_TAG_SMALL_EXPLOSION);
+                Actor_PlaySfxAtPos(&this->actor, NA_SE_IT_BOMB_EXPLOSION);
+
+                globalCtx->envCtx.unk_8C.diffuseColor1[0] = globalCtx->envCtx.unk_8C.diffuseColor1[1] =
+                    globalCtx->envCtx.unk_8C.diffuseColor1[2] = 250;
+                globalCtx->envCtx.unk_8C.ambientColor[0] = globalCtx->envCtx.unk_8C.ambientColor[1] =
+                    globalCtx->envCtx.unk_8C.ambientColor[2] = 250;
+
+                func_800DFD04(&globalCtx->mainCamera, 2, 11, 8);
+
+                ENBOMBF_GET(&this->actor) = ENBOMBF_1;
```

Comment:
wait we're using the macro this way too? lol

(I mean its fine it just looks weird to me for some reason)

Reply:
That was a find/replace accident lol. I'll put it back, good spot.